docs/usage/CATEGORY_SYNC.md

```diff
@@ -135,28 +135,40 @@ As soon, as the referenced parent Category draft is supplied to the sync, the dr
 
 ##### Syncing from a commercetools project
 
-When syncing from a source commercetools project, you can use [`mapToCategoryDrafts`](https://commercetools.github.io/commercetools-sync-java/v/4.0.1/com/commercetools/sync/categories/utils/CategoryReferenceResolutionUtils.html#mapToCategoryDrafts-java.util.List-)
-method that maps from a `Category` to `CategoryDraft` in order to make them ready for reference resolution by the sync, for example: 
+When syncing from a source commercetools project, you can use [`transformCategoryReferences`](https://commercetools.github.io/commercetools-sync-java/v/4.0.1/com/commercetools/sync/categories/service/CategoryReferenceTransformService.html#transformCategoryReferences-java.util.List-)
+the method that transforms(resolves by querying and caching key values for Ids) and maps from a `Category` to `CategoryDraft` using cache in order to make them ready for reference resolution by the sync, for example: 
 
 ````java
-// Build a CategoryQuery for fetching categories from a source CTP project with all the needed references expanded for the sync
-final CategoryQuery categoryQueryWithReferenceExpanded = CategoryReferenceResolutionUtils.buildCategoryQuery();
+// Build a CategoryQuery for fetching categories from a source CTP project without any references expanded for the sync:
+final CategoryQuery categoryQueryWithReferences = CategoryReferenceResolutionUtils.buildCategoryQuery();
 
 // Query all categories (NOTE this is just for example, please adjust your logic)
 final List<Category> categories =
     CtpQueryUtils
-        .queryAll(sphereClient, categoryQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, categoryQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the category, 
+Initialize [`CategoryReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/categories/service/CategoryReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
```

Comment:
```suggestion
For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU (Last Recently Used) based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
```

Reply:
already fixed so will resolve this conversation.